core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -548,6 +549,11 @@ public SystemSessionProperties(
                         "Rewrite semi join in filtering context to inner join",
                         featuresConfig.isRewriteFilteringSemiJoinToInnerJoin(),
                         false),
+                booleanProperty(
+                        OPTIMIZE_JOINS_BELOW_EMPTY_AGGREGATION,
+                        "Optimize joins below empty aggregations",
```

Comment:
I don't think "empty aggregation" is the best name for this. It's an aggregation with no aggregate function calls.

I would rename this to "optimize duplicate-insensitive joins"